code/modules/vehicles/quad.dm

```diff
@@ -154,6 +167,18 @@
 	var/turf/T = get_step(M, pick(throw_dirs))
 	M.throw_at(T, 1, 1, src)
 
+/obj/vehicle/train/engine/quadbike/turn_on()
+	..()
+	if(on)
+		src.visible_message("\The [src] rumbles to life.", "You hear something rumble deeply.")
+		soundloop.start()
+
+/obj/vehicle/train/engine/quadbike/turn_off()
+	if(on)
+		src.visible_message("\The [src] putters before turning off.", "You hear something putter slowly.")
+		soundloop.stop()
+	..()
```

Comment:
If this seems nonsense, it runs before on is set to 0 so that the putter message doesn't display on init

Reply:
As dumb as I find the logic that's happening on the parent call.... I think it's out of scope to ask that you make it not-dumb for the rest of vehicles/trains.